components/AttackSlot.jsx

```diff
@@ -0,0 +1,69 @@
+import styled from "styled-components";
+
+export const AttackSlot = ({ register, index, attackRemove }) => {
+  return (
+    <>
+      <StatWrapper>
+        <Stat>
+          <LabelText htmlFor={`attackSlotArray.${index}.attack-name`}>
+            Name
+          </LabelText>
+          <InputText {...register(`attackSlotArray.${index}.attack-name`)} />
+        </Stat>
+        <Stat>
+          <LabelText htmlFor={`attackSlotArray.${index}.attack-dice`}>
+            Dice
+          </LabelText>
+          <InputText {...register(`attackSlotArray.${index}.attack-dice`)} />
+        </Stat>
+        <Stat>
+          <LabelText htmlFor={`attackSlotArray.${index}.attack-range`}>
+            Range
+          </LabelText>
+          <InputText {...register(`attackSlotArray.${index}.attack-range`)} />
+        </Stat>
+        <ButtonWrapper>
+          <Button type="button" onClick={() => attackRemove(index)}>
```

Comment:
I am pointing out the above because this kinda reads unnaturally: `attackRemove(index)`